projects/systelab-components/src/lib/button/button.component.ts

```diff
@@ -0,0 +1,36 @@
+import { Component, ElementRef, EventEmitter, Input, Output } from '@angular/core';
+
+@Component({
+	selector:    'systelab-button',
+	templateUrl: 'button.component.html',
+	styles:      [`
+      :host {
+          display: inline-block;
+      }`]
+})
+export class ButtonComponent {
+
+	@Output() public action = new EventEmitter();
+
+	@Input() public disabled = false;
+
+	@Input() public type: 'regular' | 'primary' | 'warning' | 'danger' | 'success' | 'link' = 'regular';
+
+	@Input() public size: 'small' | 'medium' | 'large' = 'medium';
+
+	constructor(private element: ElementRef) {
+
+	}
+
+	public getId() {
+		return this.element.nativeElement.id;
+	}
+
+	public doClick(event: any) {
+		if (this.disabled) {
+			event.stopPropagation();
+		} else {
```

Comment:
Wouldn't be interesting a test covering this? Disabled buttons responding to click events has been an issue at some point.

Reply:
Yes it would. In fact if you look at the spec file, this is a case that is covered:

it('should not change value if icon is clicked when is disabled', () => {
		fixture.componentInstance.disabled = true;
		fixture.detectChanges();
		checkHasValue(fixture, false);
		clickOnIcon(fixture);
		checkHasValue(fixture, false);
	});

Reply:
Ok, then I don't understand that warning about line 31 not being covered...